src/language-js/printer-estree.js

```diff
@@ -4009,11 +4002,11 @@ function printFunctionParams(path, print, options, expandArg, printTypeParams) {
       parent.type === "IntersectionTypeAnnotation" ||
       (parent.type === "FunctionTypeAnnotation" &&
         parent.returnType === fun)) &&
-    printedParameters.length === 1 &&
-    printedParameters[0].node.name === null &&
-    printedParameters[0].node.typeAnnotation &&
+    parameters.length === 1 &&
+    parameters[0].name === null &&
+    parameters[0].typeAnnotation &&
     fun.typeParameters === null &&
```

Comment:
An extra check is needed here, otherwise:

**Prettier pr-9489**
[Playground link](https://deploy-preview-9489--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEMCeAHOACAjgXmwAoYALASwGclKYAncqAcwEp8A+ANwnIBMBuEABoQEDDHLRKyUAEM6dCAHcACvITSUsgDZLZaaSIBGdWWADWcGAGUMZxk2T0ArnBFwAtkbi9ePgDKyzM6yTHAAYhB0HrIwEszIILLOMBDCIKQwHtoA6hTwlHZgcNYa5BKc5WiJYJSGIIyUcHQwKqZMMcgAZjpNIgBWlAAeAEKmFlbWsh5w-oxw3b1uIIND1g7acACKzhDwi9p9IHZ0TXSJXdrK6RgMsDl8ZMgAHAAMIrcQTTmmGIm3cDOnAWIlwu3gbTEmiSlAAtFA4D4fOk6HAweRUW1Qp0kD1Dssmh5yE46K4RJQNttwQtcUsRDBZEYHrwnkgAEz00zkbQOADCEA8OJAgIArOlnE0ACqMzR4o6cVwASSgflg1jADHEAEEVdZ0JsDk0AL5GoA)
```sh
--parser flow
```

**Input:**
```jsx
type q= (this:string)=>void;
```

**Output:**
```jsx
TypeError: Cannot read property 'length' of undefined
    at Ai.each (https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:112605)
    at iterateFunctionParametersPath (https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:155225)
    at ph (https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:214765)
    at https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:256376
    at Object.print (https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:269072)
    at Ii (https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:114819)
    at https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:114280
    at Object.printComments (https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:110564)
    at e (https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:114260)
    at Ai.call (https://deploy-preview-9489--prettier.netlify.app/lib/standalone.js:15:112356)
```